Python/Include/TrialView.hpp

```diff
@@ -0,0 +1,120 @@
+#ifndef ROVER_PYTHON_TRIAL_VIEW_HPP
+#define ROVER_PYTHON_TRIAL_VIEW_HPP
+#include <functional>
+#include <string_view>
+#include <type_traits>
+#include <pybind11/pybind11.h>
+#include "Rover/ListTrial.hpp"
+#include <Rover/TrialIterator.hpp>
+#include "Rover/TrialView.hpp"
+#include "Sample.hpp"
+
+namespace Rover {
+  namespace Details {
+    class NativeTrialProxy {
+      public:
+        using Sample = PythonSample;
+        using SampleGetter = std::function<Sample (std::size_t)>;
+        using Iterator = TrialIterator<NativeTrialProxy>;
+    
+        NativeTrialProxy(const pybind11::object& view)
+          : m_view(view),
+            m_size(m_view.attr("__len__")().cast<std::size_t>()) {}
+    
+        std::size_t size() const {
+          return m_size;
+        }
+    
+        Sample operator [](std::size_t index) const {
+          auto sample = m_view.attr("__getitem__")(index);
+          auto result = SampleConverter::get_instance().cast(sample);
+          return result;
+        }
+
+        Iterator begin() const {
+          return Iterator(*this, 0);
+        }
+
+        Iterator end() const {
+          return Iterator(*this, size());
+        }
+    
+      private:
+        pybind11::object m_view;
+        std::size_t m_size;
+    };
+
+    template<typename S>
+    class PythonTrialProxy {
+      public:
+        using Sample = S;
+        using Iterator = TrialIterator<PythonTrialProxy>;
+        
+        PythonTrialProxy(const TrialView<PythonSample>& t)
+          : m_view(t) {}
+
+        std::size_t size() const {
+          return m_view.size();
+        }
+
+        Sample operator [](std::size_t index) const {
+          auto python_sample = m_view[index];
+          auto result = sample_cast<Sample>(python_sample);
+          return result;
+        }
+
+        Iterator begin() const {
+          return Iterator(*this, 0);
+        }
+
+        Iterator end() const {
+          return Iterator(*this, size());
+        }
+
+      private:
+        const TrialView<PythonSample>& m_view;
```

Comment:
Don't store references as member variables, instead store a pointer.

Reply:
Just wondering, is it a part of the coding standard or anything else? Instances of this class are never supposed to be copied.